### PR TITLE
Don't simulate mouse movement on focus change in follow_mouse = 2/3 if no_warps is false

### PR DESCRIPTION
**Pull request by antonok-edm**
#### Describe your PR, what does it fix/add?

Fixes https://github.com/hyprwm/Hyprland/issues/4949

(Technically, it looks like the original issue was fixed already in https://github.com/hyprwm/Hyprland/commit/7ea555da7, but the case described in my comment is still broken and will be fixed by this PR).

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)

I've tested all combinations of `input:follow_mouse` and `cursor:no_warps`; the behavior in this PR seems more consistent with how they're described in all cases.

#### Is it ready for merging, or does it need work?

Ready as far as I'm aware.

### CODE REVIEW

**Review by vaxerski — APPROVED** [review]
if it works, it's fine

**Comment by vaxerski** [comment]
ah clang-format missing lole

**Comment by leiserfg** [comment]
@vaxerski would you like a GitHub-action that automatically runs clang-format and makes a commit in the pr (happens every time the pr is updated)?

**Comment by vaxerski** [comment]
no

**Comment by antonok-edm** [comment]
clang-formatted

**Review by vaxerski — APPROVED** [review]
thanks!

File: src/managers/KeybindManager.cpp [hunk 1]
```diff
@@ -287,8 +287,11 @@ bool CKeybindManager::tryMoveFocusToMonitor(CMonitor* monitor) {
         return false;
     }
 
-    const auto PWORKSPACE        = g_pCompositor->m_pLastMonitor->activeWorkspace;
-    const auto PNEWMAINWORKSPACE = monitor->activeWorkspace;
+    static auto PFOLLOWMOUSE = CConfigValue<Hyprlang::INT>("input:follow_mouse");
+    static auto PNOWARPS     = CConfigValue<Hyprlang::INT>("cursor:no_warps");
+
+    const auto  PWORKSPACE        = g_pCompositor->m_pLastMonitor->activeWorkspace;
```
Comment (vaxerski) [inline src/managers/KeybindManager.cpp:293]: wait, lastmonitor can be null

Comment (antonok-edm) [reply src/managers/KeybindManager.cpp:293]: nevermind, there is one at the top of the function already

Comment (antonok-edm) [reply src/managers/KeybindManager.cpp:294]: I added it

**Review by vaxerski — APPROVED** [review]

**Pull request merged**